docs/architecture/events.rst

```diff
@@ -6,41 +6,121 @@ Events
+
+Email Event Types
+-----------------
+
++------------------------------+----------------------+----------------------------------------------------------------------------------+
+| ``payment_confirmation``     | ``PAYMENT``          | The order has been fully paid.                                                   |
++------------------------------+----------------------+----------------------------------------------------------------------------------+
+| ``shipping_confirmation``    | ``SHIPPING``         | The order has been shipped.                                                      |
++------------------------------+----------------------+----------------------------------------------------------------------------------+
+| ``tracking_updated``         | ``TRACKING_UPDATED`` | The shipping tracking number has been updated.                                   |
++------------------------------+----------------------+----------------------------------------------------------------------------------+
+| ``order_confirmation``       | ``ORDER``            | The order has been placed.                                                       |
++------------------------------+----------------------+----------------------------------------------------------------------------------+
+| ``fulfillment_confirmation`` | ``FULFILLMENT``      | One or more of the order's items have been fulfilled (both physical or digital)  |
++------------------------------+----------------------+----------------------------------------------------------------------------------+
+| ``digital_links``            | ``DIGITAL_LINKS``    | The links to the order's digital goods have been sent.                           |
++------------------------------+----------------------+----------------------------------------------------------------------------------+
+
+
+Events Design
+-------------
+
+The ``OrderEventManager`` class contains a bunch of static methods suffixed by
+``_event``. They take care of all the logic and required fields, then
+returns an instance of ``OrderEventManager`` to allow the user to commit or
+to add more events. This design has for objective to allow
+delayed (or not) event creation in the database.
+
+The main power of this design comes when one wants to create multiple
+order events in one go. Instead of generating multiple create query, one
+can simply pass the objects to django's |QuerySet.bulk_create|_ method.
+
+If one simply does not want to bulk create, delay the creation, one can
+directly run ``save()`` for the creation.
+
+.. |QuerySet.bulk_create| replace:: ``QuerySet.bulk_create``
+.. _QuerySet.bulk_create: https://docs.djangoproject.com/en/1.10/ref/models/querysets/#django.db.models.query.QuerySet.bulk_create
+
+Sending Events
```

Comment:
I wouldn't use the word "send" here as this suggests that we were using signals or some other event-based architecture underneath. We simply create objects so let's say "Creating events" and reword other parts of the docs accordingly.